Failing to create a create a config file caused a panic

Hello,

When trying to run tctl (1.16.3) on docker with restricted permission, I get this panic:

```
xxx:~$ tctl -v
creating config dir: /home/xxx/.config/temporalio
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x18b95b5]

goroutine 1 [running]:
main.main()
	/home/runner/work/tctl/tctl/cmd/tctl/main.go:39 +0x35
```

Chasing down the error, I arrived here:
https://github.com/temporalio/tctl-kit/blob/35456dd1edc915c17428588d435c16ab84d8dd62/pkg/config/config.go#L51

If an error happens when we create the file, it's swallowed. It's missing an `if err != nil { return nil, err }`.

--- a/pkg/config/config.go
+++ b/pkg/config/config.go
@@ -23,6 +23,7 @@
 package config
 
 import (
+	"errors"
 	"fmt"
 	"os"
 	"path/filepath"
@@ -48,9 +49,11 @@ func NewConfig(appName, configName string) (*Config, error) {
 
 	dpath = filepath.Join(dpath, ".config", appName)
 
-	mkfile(dpath, configName+".yaml")
+	cfgPath, err := mkfile(dpath, configName+".yaml")
+	if err != nil {
+		return nil, err
+	}
 
-	cfgPath := filepath.Join(dpath, configName+".yaml")
 	cfg, err := readConfig(cfgPath)
 	if err != nil {
 		return nil, err
@@ -164,7 +167,12 @@ func mkfile(dirPath, filename string) (string, error) {
 
 	fpath := filepath.Join(dirPath, filename)
 
-	if _, err := os.Stat(fpath); err != nil {
+	_, err := os.Stat(fpath)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return "", err
+		}
+
 		fmt.Printf("creating config file: %v\n", fpath)
 		file, err := os.Create(fpath)
 		if err == nil {

--- a/pkg/config/config_test.go
+++ b/pkg/config/config_test.go
@@ -38,6 +38,16 @@ const (
 	cfgFile = "test-tctl-kit-config"
 )
 
+func TestNewConfigPermissionDenied(t *testing.T) {
+	expectedError := errors.New("open /tmp.yaml: permission denied")
+
+	_, err := config.NewConfig(appName, "../../../../../../../../../../../../tmp")
+
+	if assert.Error(t, err) {
+		assert.Equal(t, expectedError.Error(), err.Error())
+	}
+}
+
 func TestNewConfigCreatesFile(t *testing.T) {
 	path := getConfigPath(t)
 

